src/sardana/macroserver/msmacromanager.py

```diff
@@ -1290,9 +1319,56 @@ def __runXMLMacro(self, xml):
     _runXMLMacro = __runXMLMacro
 
     def runMacro(self, macro_obj):
+
         name = macro_obj._getName()
         desc = macro_obj._getDescription()
         door = self.door
+        macro_obj = macro_obj
+
+        if macro_obj.getParentMacro() is None:
+            try:
+                logging_onoff = macro_obj.getEnv("LogMacro")
+            except:
+                logging_onoff = False
+            if logging_onoff:
+                try:
+                    self.logging_mode = macro_obj.getEnv("LogMacroMode")
+                except:
+                    self.logging_mode = 0
+                try:
+                    logging_path = macro_obj.getEnv("LogMacroDir")
+                except:
+                    macro_obj.setEnv("LogMacroDir", "/tmp")
+                    logging_path = macro_obj.getEnv("LogMacroDir")
+                    
+                log_file = logging_path + "/spock_session_" + socket.gethostname() +".log"
```

Comment:
Here I have two comments:
- I would not use "spock" in the file name - other clients e.g. macroexecutor could also take a profit from this feature
- maybe in the file name we could use the door name instead of the hostname. Otherwise if we run two clients, each on a separate door this would log to the same file.

Reply:
I posted #689 that fixed the issue with door's name.